alica_engine/src/engine/AlicaTimer.cpp

```diff
@@ -0,0 +1,68 @@
+#include "engine/AlicaTimer.h"
+
+namespace alica
+{
+
+class AlicaSystemTimer::AlicaSystemTimerImpl
+{
+public:
+    using TimerCb = std::function<void()>;
+
+    AlicaSystemTimerImpl(TimerCb&& userCb, alica::AlicaTime period)
+            : _userCb(std::move(userCb))
+            , _period(period.inMilliseconds())
+            , _isActive(false)
+    {
+    }
+
+    ~AlicaSystemTimerImpl()
+    {
+        if (_isActive) {
+            stop();
+        }
+    }
+
+    void start()
+    {
+        if (_isActive) {
+            stop();
+        }
+        _isActive = true;
+        _thread = std::thread([this]() {
+            while (_isActive) {
+                _userCb();
+                int64_t sleep_duration = _period;
+                while ((sleep_duration > 0) && _isActive) {
```

Comment:
why can't the thread sleep for the entire duration at once?

Reply:
This is a trick to help shutdown more easily on ctrl c mainly.  Not completely sure if it's necessary

Reply:
ah ok, its fine then. Can maybe add a comment to clarify